webapp/channels/src/components/post_view/message_attachments/action_button/action_button.tsx

```diff
@@ -31,47 +38,44 @@ export default class ActionButton extends React.PureComponent<Props> {
             primary: theme.buttonBg,
             success: '#339970',
         } as Record<string, string>;
-    }
+    };
 
-    render() {
-        const {action, handleAction, disabled, theme} = this.props;
-        let hexColor: string | null | undefined;
+    let hexColor: string | null | undefined;
 
-        if (action.style) {
-            const STATUS_COLORS = this.getStatusColors(theme);
-            hexColor =
-                STATUS_COLORS[action.style] ||
-                theme[action.style] ||
-                (action.style.match('^#(?:[0-9a-fA-F]{3}){1,2}$') && action.style);
-        }
+    if (action.style) {
+        const STATUS_COLORS = getStatusColors(theme);
+        hexColor =
+            STATUS_COLORS[action.style] ||
+            theme[action.style] ||
+            (action.style.match('^#(?:[0-9a-fA-F]{3}){1,2}$') && action.style);
+    }
 
-        return (
-            <ActionBtn
-                data-action-id={action.id}
-                data-action-cookie={action.cookie}
-                disabled={disabled}
-                key={action.id}
-                onClick={(e) => handleAction(e, this.props.action.options)}
-                className='btn btn-sm'
-                hexColor={hexColor}
+    return (
+        <ActionBtn
+            data-action-id={action.id}
+            data-action-cookie={action.cookie}
+            disabled={disabled}
+            key={action.id}
+            onClick={(e) => handleAction(e, action.options)}
+            className='btn btn-sm'
+            hexColor={hexColor}
+        >
+            <LoadingWrapper
+                loading={actionExecuting}
+                text={actionExecutingMessage}
             >
-                <LoadingWrapper
-                    loading={this.props.actionExecuting}
-                    text={this.props.actionExecutingMessage}
-                >
-                    <Markdown
-                        message={action.name}
-                        options={{
-                            mentionHighlight: false,
-                            markdown: false,
-                            autolinkedUrlSchemes: [],
-                        }}
-                    />
-                </LoadingWrapper>
-            </ActionBtn>
-        );
-    }
-}
+                <Markdown
+                    message={action.name}
+                    options={{
+                        mentionHighlight: false,
+                        markdown: false,
+                        autolinkedUrlSchemes: [],
+                    }}
```

Comment:
Since this is the same on every render, let's move it also outside of the component, so we don't pass a new object on every render.